search-provider/eks-metadata-provider-dbus.xml

```diff
@@ -0,0 +1,137 @@
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General
+ Public License along with this library; if not, see <http://www.gnu.org/licenses/>.
+-->
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <interface name="com.endlessm.ContentMetadata">
+    <!--
+        Query:
+        @Query: A dictionary describing the query to be made, with the following
+                optional parameters. Specifying any parameter that is not
+                a member of this list is an error. No new parameters will
+                be added or removed from this interface during its lifetime.
+
+                "search-terms": A string (s) with the free-text search terms
+                                to use during the query. If the parameter
+                                is not specified, filtering will only
+                                be done against tags-match-any and tags-match-all.
+                "tags-match-any": A strv (as) with a list of tags that any
+                                  matching content objects must have at least
+                                  one of.
+                "tags-match-all": A strv (as) with a list of tags that any
+                                  matching content objects must have all of.
+                "limit": An integer indicating the number of results at most
+                         to return. If the parameter is not specifed there
+                         will be no limit.
+                "offset": An integer specifying the "offset" into the query
+                          results from the point which it start returning results.
+                          This also changes the index of the last result if "limit"
+                          was specified. For instance, if "offset" was 1 and
+                          "limit" was 5, then the 2nd to 6th results will
+                          be returned, inclusive.
+                "sort": A string specifying what the query should be sorted by,
+                        valid values are:
+
+                            "relevance": Sort by relevance ranking, with exact
+                                         title matches weighted most heavily.
+                            "sequence-number": Sort by article page ranking.
+                            "date": Sort by date of publication.
+                            "alphabetical" Sort alphabetically.
+
+                "order": A string specifying how sorted query results should
+                         be ordered, valid values are:
+
+                             "ascending": Least-ranked to most-ranked.
+                             "descending": Most-ranked to least-ranked.
```

Comment:
Last minute thought: it may be better to have this interface take an array of queries (or to add another method taking such an array). In the companion app services I already have a case where we have to do two queries (one to get the sets and one to get the relevant metadata) and I dont want to be stuck in a situation where we are able to optimize that on the xapian side but are stuck due to the design of this interface. I'll adjust tonight.

Reply:
OK, I'll try to get my companion app service setup and working.

Reply:
@smspillaz To be honest, that sounds a little bit like a premature optimization at this point and I'd rather get something working in the simple case for this first version of the interface first...

Reply:
He does have a point that we should get the API right ... I'd like to avoid MultiEknServices2 if we can.

Reply:
Yes, I don't think we should actually implement that support, but we should allow arrays of queries and results in the API. That should be a small change relative to this pull request.

I think until the time we need that optimization, the API method should only support a query array with one element, and throw otherwise.

Reply:
Yeah I think this is a good compromise. We can always add the actual complex parts later.